src/decode-ethernet.c

```diff
@@ -85,6 +85,12 @@ int DecodeEthernet(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p,
             DecodeMPLS(tv, dtv, p, pkt + ETHERNET_HEADER_LEN,
                        len - ETHERNET_HEADER_LEN, pq);
             break;
+        case ETHERNET_TYPE_DCE:
+            if (len > ETHERNET_DCE_HEADER_LEN) {
```

Comment:
I think I would like to see an event if this condition is false. Otherwise we'll completely silently ignore packets w/o any logging or possibility to match on the packet.